### DISCUSSION TITLE
More convenient way to make translations for actions or spices in general

### DISCUSSION THREAD

**Discussion by eco-star** [linuxmint/discussions#372] [Ideas]
Hi Guys,

I just took a look at the cinnamon-spices-actions repo and noticed that there is a lot to do in translation. I was happy about it and wanted to provide translations for german and brazilian portuguese (pt-BR) but I failed at the instructions listed down below.

[](https://github.com/linuxmint/cinnamon-spices-actions/tree/master)

To be honest: I am not that familiar with git yet and I am making my first steps, so please do not judge me :)

Is there a noob-friendly instruction which I can follow step-by-step and from "zero-to-hero" to provide some translations?
Or is there a way to provide translations via Launchpad? I am already using Launchpad to provide translations to Linux Mint.

Thank you for all your work and improvements in Mint and I am happy to help here as good as I can.

Regards and happy eastern :)

**Comment by rcalixte** [comment]
Thank you @eco-star! In terms of the Spices, most will have an individual .pot file in a subdirectory, along the lines of UUID/files/UUID/po/UUID.pot. This is the translation template file. In that same directory, there are the .po files. These are the translations where the language code is the file name. When it comes to tooling, there is software called Poedit that can be installed via local repository or Flatpak. This is what most translators use. Poedit can create new translation files using the .pot templates or it can edit existing translation files. The interface also has many options to assist in the translation creation process.

The next part, as you indicated, is how to feed these translation files to GitHub. There is a command line for `git` but GitHub's UI has also improved dramatically in this area as well. The core concepts here that matter are:

* [Fork(s)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/fork-a-repo) (creating a copy of the repository, e.g. cinnamon-spices-applets)
* [Branch(es)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-and-deleting-branches-within-your-repository) (creating a working area where you can add/edit one or many files, named whatever you want)
* [Pull Request(s)](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request) (creating a request to a repository to have your changes added)

If it's something you want to start doing, you can start small, working with one file at a time. Then you can work in batches after that if you'd like once you get comfortable. Once you have a few repetitions, I think you'll find the process to hopefully not be too cumbersome. As always, we haev a strong community here to assist if needed as well!

**Reply by eco-star** [reply]
Hi @rcalixte thank you for your advise with Poedit and your instruction. I will try it again with more time and patience :)

**Comment by rcalixte** [comment]
As a tangent to your post (and why I wanted to add it separately), I've noticed that the workflow for translations for Spices isn't the easiest for everyone. I was contemplating the idea of a Discussion post (pretty recently actually, so this is very timely!). I wanted to float the idea of something like Weblate for translations as I have seen that in use by many projects and it seems to be becoming something of the standard open source platform for translations. I don't know if you have experience with it. The goal for something like that would be so that the process is similar to your experience with Launchpad. If this sounds like an interesting idea, I can go ahead and work on creating that Discussion thread to see what interest exists in the community.

**Reply by eco-star** [reply]
Yes, it sounds like an awesome idea :)
In addition to that this tool could be used as a central translation platform in general for everything related to Mint, even for the official webpage if that is possible :)

**Comment by klangman** [comment]
Since I am only fluent in English (and some might say not even that) I have never tried to translate anything, so I don't really know the process that well. 

But I was thinking that the new hotkeys options that I just now added to the PanelTranslator applet might help speed up translating work a bit. If people are actually typing in translations into poedit, then using PanelTransltor@klangman can maybe help. You can setup a hotkey (i.e. Ctrl+Super+T) in the configuration to translate the current selection and copy the translation to the clipboard. Then you can setup PanelTranslator for English to the target language translating. Now in poedit, you simply select the text to translate (Ctrl+A), type the hotkey you defined in PanelTranslator (Ctrl+Super+T), then paste (Ctrl+V) into the translation entry field in poedit. Now you have your translation which you can read and edit if the translation is not perfect already.

Since I don't do this work, maybe there already exist better automation options for this work??

**Reply by rcalixte** [reply]
> Since I don't do this work, maybe there already exist better automation options for this work??

From ChatGPT to other mainstream tools that now exist, the challenge of translation in general is that languages consist of a tremendous amount of nuance. The specific challenge (and this has been attempted before with non-optimal results) is that the translations for Cinnamon-related strings require context and are largely focused around a more technical application than what any of the mainstream tools can handle at the moment.
